tmva/tmva/src/MethodDL.cxx

```diff
@@ -467,11 +467,11 @@ void MethodDL::ParseInputLayout()
    // when we will support 3D convolutions we would need to add extra 1's
    if (inputShape.size() == 2) {
       // case of dense layer where only width is specified
-      inputShape.insert(inputShape.begin() + 1, {1,1});
+      inputShape = {inputShape[0], 1, 1, inputShape[1]};
```

Comment:
Do you know why this gives a warning?

Reply:
No, sorry. It could even be a compiler bug, or maybe because the iterator is invalidated after the first insertion in case a re-allocation happens.

In any case, I think the new formulation of the logic is cleaner, so I wouldn't try harder to really understand this warning.

Reply:
ok, fine!